### PR TITLE
watch vrt nu abroad: support for roaming in european union

### PR DESCRIPTION
**Pull request by mediaminister**
I implemented support for watching VRT NU abroad. If the user has a validated Belgian mobile phone number in his VRT account, the user can watch all tv-programmes in the European Union.

Otherwise, the user get the following message: 
> "This program can not be played. To view this program outside of Belgium, you first have to validate a Belgian mobile phone number on the VRT NU website."

### CODE REVIEW

File: plugin.video.vrt.nu/resources/lib/vrtplayer/urltostreamservice.py [hunk 1]
```diff
@@ -75,33 +79,64 @@ def _get_playertoken(self, token_url, xvrttoken=None):
...
                 return self._get_new_xvrttoken(path)
         else:
             title = self._kodi_wrapper.get_localized_string(32051)
             message = self._kodi_wrapper.get_localized_string(32052)
             self._kodi_wrapper.show_ok_dialog(title, message)
 
-    def _get_xvrttoken(self):
-        tokenfile = self._kodi_wrapper.get_userdata_path() + 'XVRTToken'
+    def _get_xvrttoken(self, tokenfile=None):
+        if tokenfile is None:
```
Comment (pietje666) [inline plugin.video.vrt.nu/resources/lib/vrtplayer/urltostreamservice.py]: rewrite 
        if tokenfile is None:
            tokenfile = self._kodi_wrapper.get_userdata_path() + 'XVRTToken'
        else:
            tokenfile = self._kodi_wrapper.get_userdata_path() + tokenfile

as tokenfile = self._kodi_wrapper.get_userdata_path() + (tokenfile or 'XVRTToken')

File: plugin.video.vrt.nu/resources/lib/vrtplayer/urltostreamservice.py [hunk 2]
```diff
@@ -150,39 +185,54 @@ def _get_license_key(self, key_url, key_type='R', key_headers=None, key_value=No
             return ''.join((key_url, '|', header.strip('&'), '|', key_value, '|'))
 
 
-    def get_stream_from_url(self, url):
-        url = urljoin(self._vrt_base, url)
-        html_page = requests.get(url).text
+    def get_stream_from_url(self, url, xvrttoken=None):
```
Comment (pietje666) [inline plugin.video.vrt.nu/resources/lib/vrtplayer/urltostreamservice.py]: this method is becomming quite bulky cant we split it up in some more logical pieces? To improve readability and maintenance

**Comment by pietje666** [comment]
Thx for real bugfix

**Comment by mediaminister** [comment]
You're welcome, I think the code is ready to be merged now.

**Comment by pietje666** [comment]
yes but will refactor some more in my own master branch :)

**Pull request merged**